src/utils/config_utils/tests/config_utils_ut.cpp

```diff
@@ -373,15 +374,405 @@ static const char* invalidConfigContentDownloadTimeout =
+ * The actual MQTT broker connection settings are defined in src/extensions/agent_modules/communication_modules/inc.
+ * This data structure is used to compare the values read from the config file with the expected values.
+ */
+typedef struct _mock_adps_settings
+{
+  char* idScope;           /*< Device Provisioning Service Id Scope */
+  char* registrationId;    /*< Device Provisioning Service Registration Id */
+  char* dpsApiVersion;     /*< Device Provisioning Service API Version */
+  char* caFile;            /*< Path to a PEM file with the chain required to trust the TLS endpoint certificate */
+  char* certFile;          /*< Path to a PEM file to establish X509 client authentication */
+  char* clientId;          /*< MQTT Client Id */
+  char* hostname;          /*< FQDM to the MQTT Broker endpoint, eg: mybroker.mydomain.com */
+  char* keyFile;           /*< Path to a KEY file to establish X509 client authentication */
+  char* keyFilePassword;   /*< Password (aka pass-phrase) to protect the key file */
+  char* password;          /*< MQTT Password to authenticate the connection */
+  char* username;          /*< MQTT Username to authenticate the connection */
+  unsigned int keepAliveInSeconds; /*< Seconds to send the ping to keep the connection open */
+  unsigned int tcpPort;    /*< TCP port to access the endpoint eg: 8883 */
+  bool cleanSession;       /*< MQTT Clean Session, might require to set the ClientId [existing sessions are not supported now] */
+  bool useTLS;             /*< Disable TLS negotiation (not recommended for production)*/
+} Mock_ADPS_Settings;
+
+#define DEFAULT_TCP_PORT 8883
+#define DEFAULT_KEEP_ALIVE_IN_SECONDS 30
+#define DEFAULT_USE_TLS true
+#define DEFAULT_CLEAN_SESSION false
+#define DEFAULT_MQTTBROKER_CLEAN_SESSION true
+
+static void _free_mock_adps_settings(Mock_ADPS_Settings* settings)
+{
+    if (settings == NULL)
+    {
+        return;
+    }
+
+    free(settings->idScope);
+    free(settings->registrationId);
+    free(settings->dpsApiVersion);
+
+    free(settings->hostname);
+    free(settings->clientId);
+    free(settings->username);
+    free(settings->password);
+    free(settings->caFile);
+    free(settings->certFile);
+    free(settings->keyFile);
+    free(settings->keyFilePassword);
+    memset(settings, 0, sizeof(*settings));
+}
+
+static bool _read_mock_adps_settings(Mock_ADPS_Settings* settings)
+{
+    bool result = false;
+    const ADUC_ConfigInfo* config = NULL;
+    const ADUC_AgentInfo* agent_info = NULL;
+
+    if (settings == NULL)
+    {
+        goto done;
+    }
+
+    memset(settings, 0, sizeof(*settings));
+
+    config = ADUC_ConfigInfo_GetInstance();
+    if (config == NULL) {
+        goto done;
+    }
+
+    agent_info = ADUC_ConfigInfo_GetAgent(config, 0);
+
+    // Currently only support 'ADPS2/MQTT' connection data.
+    if (strcmp(agent_info->connectionType, ADUC_CONNECTION_TYPE_ADPS2_MQTT) != 0)
+    {
+        printf("Connection type is not ADPS2/MQTT");
```

Comment:
remove all `printf` in this file and use REQUIRE instead? e.g. REQUIRE(strcmp(...) == 0) here

Reply:
Removed all printf from mock & test functions